### PR TITLE
perf(orama): improve speed of heavy tasks

### PR DESCRIPTION
**Pull request by ilteoood**
With this PR I would like to make some code changes in order to improve Orama's speed during heavy operations, like:
- for loops
- extraction of the same data
- function called with a high number of parameters
- new object creation only when needed
- usage of optional chaining

Additionally, since I don't know the details of Orama's codebase, I have inserted a couple of comments for future improvements that I can do in this PR or in another one.

Note for the reviewers: the only changes I made are in the `orama` package. Everything else is related to formatting fixes made by the biome upgrade

### CODE REVIEW

File: packages/orama/src/components/facets.ts [hunk 1]
```diff
@@ -99,16 +98,18 @@ export async function getFacets<T extends AnyOrama>(
     }
   }
 
+  // We are looping again with the same previous keys, should we creat a single loop instead?
```
Comment (ilteoood) [inline packages/orama/src/components/facets.ts:101]:
```suggestion
  // TODO: We are looping again with the same previous keys, should we creat a single loop instead?
```

**Comment by micheleriva** [comment]
Wow! Thank you Matteo! Please make sure to format the code correctly by running `pnpm format` before committing 🙏

Will review this ASAP

**Comment by ilteoood** [comment]
@micheleriva version 1.4.1 of biome was not working on GitHub codespaces. I upgraded it to the latest one (1.6.4) and executed the `format` script but...the number of changed files slightly increased 🥲

**Comment by allevo** [comment]
I see a performance improvement of 5% in some cases during the search.
I run the scripts [here](https://github.com/allevo/orama-version-comparator).

**Comment by micheleriva** [comment]
This is awesome

**Review by micheleriva — APPROVED** [review]

**Pull request merged**